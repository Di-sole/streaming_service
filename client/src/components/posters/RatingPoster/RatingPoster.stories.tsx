import { Meta, StoryObj } from '@storybook/react';
import './RatingPoster.module.scss';
import Poster from './RatingPoster';
import IMovie from '@/models/IMovie';

const meta: Meta<typeof Poster> = {
    title: 'Poster/RatingPoster',
    component: Poster,
    decorators: [
        (Story) => (
            <div
                style={{
                    width: '1000px',
                }}
            >
                <Story />
            </div>
        ),
    ],
};

export default meta;

type Story = StoryObj<typeof Poster>;

interface IRatingMovie extends IMovie {
    logo?: string;
    place?: number;
}

const ratingMovies: IRatingMovie = {
    id: 1,
    type: '',
    name: '',
    enName: '',
    posterUrl: require('../../../testAsserts/img/kingfisher.jpg'),
    posterPreviewURL: '',
    logo: require('../../../testAsserts/img/kingfisher.png'),
    year: 2022,
    description: '',
    shortDescription: '',
    ageRating: 18,
    ratingKp: 8,
    votesKp: 1000,
    movieLength: 120,
    genres: [],
    countries: [],
    persons: [],
    place: 1,
    trailer: '',
    watchingWithMovie: [],
    comments: [],
};

export const RatingPoster: Story = {
    args: {
        movie: ratingMovies,
    },
};
