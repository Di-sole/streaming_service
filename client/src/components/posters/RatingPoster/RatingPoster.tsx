import Image from 'next/image';
import Link from 'next/link';
import style from './RatingPoster.module.scss';
import IMovie from '@/models/IMovie';
import { IRatingMovie } from './ratingMovies.data';

interface RatingPosterProps {
    movie: IRatingMovie;
    className: string;
}

const RatingPoster = ({ movie, className }: RatingPosterProps) => {
    return (
        <div className={[style.container, className].join(' ')}>
            <Link href="/">
                <div className={style.container__imageWrapper}>
                    <Image className={style.container__image} fill src={movie.posterUrl} alt={movie.name} />
                    <div className={style.container__fadeArea}></div>
                </div>
                <div className={style.container__content}>
                    <div className={style.container__fadeArea}></div>

                    <div className={style.container__logoWrapper}>
                        <Image
                            className={style.container__logo}
                            width={153}
                            height={40}
                            src={movie.logo!}
                            alt={movie.name}
                        />
                    </div>

                    <div className={style.container__numberWrapper}>
                        <Image
                            className={style.container__placeNumber}
                            width={48}
                            height={66}
                            src={`https://solea-parent.dfs.ivi.ru/picture/bypass/number${movie.place}.svg`}
                            alt={`${movie.place}`}
                        />
                    </div>
                </div>
            </Link>
        </div>
    );
};

export default RatingPoster;
