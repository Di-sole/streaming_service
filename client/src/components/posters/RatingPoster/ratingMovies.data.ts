import IMovie from '@/models/IMovie';

export interface IRatingMovie extends IMovie {
    logo?: string;
    place?: number;
}

export const ratingMovies: IRatingMovie[] = [
    {
        id: 1,
        type: '',
        name: 'зимородок',
        enName: '',
        posterUrl: 'https://thumbs.dfs.ivi.ru/storage6/contents/e/0/08c76a2df45108ec28af6300cc5b93.jpg/304x620//?q=85',
        posterPreviewURL: '',
        logo: 'https://thumbs.dfs.ivi.ru/storage4/contents/b/3/826ab6a9a975f9919837f0d17ba642.png/x200/?q=85',
        year: 2023,
        description: '',
        shortDescription: '',
        ageRating: 18,
        ratingKp: 8,
        votesKp: 1000,
        movieLength: 120,
        genres: [],
        countries: [],
        persons: [],
        place: 1,
        trailer: '',
        watchingWithMovie: [],
        comments: [],
    },
    {
        id: 1,
        type: '',
        name: 'против всех',
        enName: '',
        posterUrl: 'https://thumbs.dfs.ivi.ru/storage26/contents/a/1/56b0cedeb8dd5890ee46ab947c1207.jpg/304x620//?q=85',
        posterPreviewURL: '',
        logo: 'https://thumbs.dfs.ivi.ru/storage2/contents/5/0/5a82ac6f5942eeea490540feca0a62.png/x200/?q=85',
        year: 2023,
        description: '',
        shortDescription: '',
        ageRating: 18,
        ratingKp: 8,
        votesKp: 1000,
        movieLength: 120,
        genres: [],
        countries: [],
        persons: [],
        place: 2,
        trailer: '',
        watchingWithMovie: [],
        comments: [],
    },
    {
        id: 1,
        type: '',
        name: 'семья',
        enName: '',
        posterUrl: 'https://thumbs.dfs.ivi.ru/storage28/contents/d/e/f3afc43a0709ea1ebf35cdf142cc46.jpg/304x620//?q=85',
        posterPreviewURL: '',
        logo: 'https://thumbs.dfs.ivi.ru/storage9/contents/d/e/14ad136916cb3797041ef18a0b6149.png/x200/?q=85',
        year: 2023,
        description: '',
        shortDescription: '',
        ageRating: 18,
        ratingKp: 8,
        votesKp: 1000,
        movieLength: 120,
        genres: [],
        countries: [],
        persons: [],
        place: 3,
        trailer: '',
        watchingWithMovie: [],
        comments: [],
    },
    {
        id: 1,
        type: '',
        name: 'шпион',
        enName: '',
        posterUrl: 'https://thumbs.dfs.ivi.ru/storage28/contents/f/2/7e50d51661b729863f8584ee559242.jpg/304x620//?q=85',
        posterPreviewURL: '',
        logo: 'https://thumbs.dfs.ivi.ru/storage9/contents/b/6/356258bbe7c5ba5b5b40251be3d48f.png/x200/?q=85',
        year: 2023,
        description: '',
        shortDescription: '',
        ageRating: 18,
        ratingKp: 8,
        votesKp: 1000,
        movieLength: 120,
        genres: [],
        countries: [],
        persons: [],
        place: 4,
        trailer: '',
        watchingWithMovie: [],
        comments: [],
    },
    {
        id: 1,
        type: '',
        name: 'и снова здравствуйте',
        enName: '',
        posterUrl: 'https://thumbs.dfs.ivi.ru/storage4/contents/2/c/3b8e2a957b926355725e1237e188bf.jpg/304x620//?q=85',
        posterPreviewURL: '',
        logo: 'https://thumbs.dfs.ivi.ru/storage9/contents/7/a/f1fdf81472f76cc5950e0f4ef30060.png/x200/?q=85',
        year: 2023,
        description: '',
        shortDescription: '',
        ageRating: 18,
        ratingKp: 8,
        votesKp: 1000,
        movieLength: 120,
        genres: [],
        countries: [],
        persons: [],
        place: 5,
        trailer: '',
        watchingWithMovie: [],
        comments: [],
    },
    {
        id: 1,
        type: '',
        name: 'праведник',
        enName: '',
        posterUrl: 'https://thumbs.dfs.ivi.ru/storage3/contents/0/9/09f0b7761bcd8ceb9c42bb93f4806c.jpg/304x620//?q=85',
        posterPreviewURL: '',
        logo: 'https://thumbs.dfs.ivi.ru/storage28/contents/e/e/2b49a5aa4d8ab07304588192853bee.png/x200/?q=85',
        year: 2023,
        description: '',
        shortDescription: '',
        ageRating: 18,
        ratingKp: 8,
        votesKp: 1000,
        movieLength: 120,
        genres: [],
        countries: [],
        persons: [],
        place: 6,
        trailer: '',
        watchingWithMovie: [],
        comments: [],
    },
    {
        id: 1,
        type: '',
        name: 'здоровый человек',
        enName: '',
        posterUrl: 'https://thumbs.dfs.ivi.ru/storage5/contents/9/9/f92d6d3965cfcb7b7e1878eaa56f05.jpg/304x620//?q=85',
        posterPreviewURL: '',
        logo: 'https://thumbs.dfs.ivi.ru/storage30/contents/a/a/c55629c1cb82b0ac7c0d9aca539d89.png/x200/?q=85',
        year: 2023,
        description: '',
        shortDescription: '',
        ageRating: 18,
        ratingKp: 8,
        votesKp: 1000,
        movieLength: 120,
        genres: [],
        countries: [],
        persons: [],
        place: 7,
        trailer: '',
        watchingWithMovie: [],
        comments: [],
    },
    {
        id: 1,
        type: '',
        name: 'операция фортуна',
        enName: '',
        posterUrl: 'https://thumbs.dfs.ivi.ru/storage9/contents/8/0/86ca0a529c83b35c0b0f2b29a0fa57.jpg/304x620//?q=85',
        posterPreviewURL: '',
        logo: 'https://thumbs.dfs.ivi.ru/storage9/contents/0/7/c29756fa4d94a6ba334836e7820b3a.png/x200/?q=85',
        year: 2023,
        description: '',
        shortDescription: '',
        ageRating: 18,
        ratingKp: 8,
        votesKp: 1000,
        movieLength: 120,
        genres: [],
        countries: [],
        persons: [],
        place: 8,
        trailer: '',
        watchingWithMovie: [],
        comments: [],
    },
    {
        id: 1,
        type: '',
        name: 'непослушная',
        enName: '',
        posterUrl: 'https://thumbs.dfs.ivi.ru/storage29/contents/1/9/2475ddbdc646eb062cc61cb73db2a8.jpg/304x620//?q=85',
        posterPreviewURL: '',
        logo: 'https://thumbs.dfs.ivi.ru/storage28/contents/f/4/9d0d6b3e5fd0cd0a35080f6950d1c6.png/x200/?q=85',
        year: 2023,
        description: '',
        shortDescription: '',
        ageRating: 18,
        ratingKp: 8,
        votesKp: 1000,
        movieLength: 120,
        genres: [],
        countries: [],
        persons: [],
        place: 9,
        trailer: '',
        watchingWithMovie: [],
        comments: [],
    },
    {
        id: 1,
        type: '',
        name: 'постучись в мою дверь',
        enName: '',
        posterUrl: 'https://thumbs.dfs.ivi.ru/storage33/contents/b/2/661be4b1a65157bb5440ffc3bbeb06.jpg/304x620//?q=85',
        posterPreviewURL: '',
        logo: 'https://thumbs.dfs.ivi.ru/storage28/contents/4/8/c0919a0f0fce59bcaaa0ca5dea3dec.png/x200/?q=85',
        year: 2023,
        description: '',
        shortDescription: '',
        ageRating: 18,
        ratingKp: 8,
        votesKp: 1000,
        movieLength: 120,
        genres: [],
        countries: [],
        persons: [],
        place: 10,
        trailer: '',
        watchingWithMovie: [],
        comments: [],
    },
];
