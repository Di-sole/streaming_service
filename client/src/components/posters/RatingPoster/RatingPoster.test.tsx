import { render, screen } from '@testing-library/react';
import RatingPoster from './RatingPoster';
import { ratingMovieTest } from '../../../testAsserts/testItems';

jest.mock('next/router', () => ({
    useRouter() {
        return {
            pathname: '',
        };
    },
}));

describe('RATINGPOSTER SNAPSHOTS TESTS', () => {
    test('RatingPoster should not be changed', () => {
        render(<RatingPoster movie={ratingMovieTest} className="ratingItem" />);

        const id = screen.getByTestId('ratingPoster');
        expect(id).toMatchSnapshot();
    });
});
