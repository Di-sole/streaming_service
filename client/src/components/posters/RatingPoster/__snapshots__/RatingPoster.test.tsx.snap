// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RATINGPOSTER SNAPSHOTS TESTS RatingPoster should not be changed 1`] = `
<div
  class="container ratingItem"
  data-testid="ratingPoster"
>
  <a
    href="/movies/1"
  >
    <div
      class="container__imageWrapper"
    >
      <img
        alt="зимородок"
        class="container__image"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="100vw"
        src="/_next/image?url=https%3A%2F%2Fthumbs.dfs.ivi.ru%2Fstorage6%2Fcontents%2Fe%2F0%2F08c76a2df45108ec28af6300cc5b93.jpg%2F304x620%2F%2F%3Fq%3D85&w=3840&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fthumbs.dfs.ivi.ru%2Fstorage6%2Fcontents%2Fe%2F0%2F08c76a2df45108ec28af6300cc5b93.jpg%2F304x620%2F%2F%3Fq%3D85&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fthumbs.dfs.ivi.ru%2Fstorage6%2Fcontents%2Fe%2F0%2F08c76a2df45108ec28af6300cc5b93.jpg%2F304x620%2F%2F%3Fq%3D85&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fthumbs.dfs.ivi.ru%2Fstorage6%2Fcontents%2Fe%2F0%2F08c76a2df45108ec28af6300cc5b93.jpg%2F304x620%2F%2F%3Fq%3D85&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fthumbs.dfs.ivi.ru%2Fstorage6%2Fcontents%2Fe%2F0%2F08c76a2df45108ec28af6300cc5b93.jpg%2F304x620%2F%2F%3Fq%3D85&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fthumbs.dfs.ivi.ru%2Fstorage6%2Fcontents%2Fe%2F0%2F08c76a2df45108ec28af6300cc5b93.jpg%2F304x620%2F%2F%3Fq%3D85&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fthumbs.dfs.ivi.ru%2Fstorage6%2Fcontents%2Fe%2F0%2F08c76a2df45108ec28af6300cc5b93.jpg%2F304x620%2F%2F%3Fq%3D85&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fthumbs.dfs.ivi.ru%2Fstorage6%2Fcontents%2Fe%2F0%2F08c76a2df45108ec28af6300cc5b93.jpg%2F304x620%2F%2F%3Fq%3D85&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fthumbs.dfs.ivi.ru%2Fstorage6%2Fcontents%2Fe%2F0%2F08c76a2df45108ec28af6300cc5b93.jpg%2F304x620%2F%2F%3Fq%3D85&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
      <div
        class="container__fadeArea"
      />
    </div>
    <div
      class="container__content"
    >
      <div
        class="container__fadeArea"
      />
      <div
        class="container__logoWrapper"
      >
        <img
          alt="зимородок"
          class="container__logo"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fthumbs.dfs.ivi.ru%2Fstorage4%2Fcontents%2Fb%2F3%2F826ab6a9a975f9919837f0d17ba642.png%2Fx200%2F%3Fq%3D85&w=384&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fthumbs.dfs.ivi.ru%2Fstorage4%2Fcontents%2Fb%2F3%2F826ab6a9a975f9919837f0d17ba642.png%2Fx200%2F%3Fq%3D85&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fthumbs.dfs.ivi.ru%2Fstorage4%2Fcontents%2Fb%2F3%2F826ab6a9a975f9919837f0d17ba642.png%2Fx200%2F%3Fq%3D85&w=384&q=75 2x"
          style="color: transparent;"
          width="153"
        />
      </div>
      <div
        class="container__numberWrapper"
      >
        <img
          alt="1"
          class="container__placeNumber"
          data-nimg="1"
          decoding="async"
          height="66"
          loading="lazy"
          src="https://solea-parent.dfs.ivi.ru/picture/bypass/number1.svg"
          style="color: transparent;"
          width="48"
        />
      </div>
    </div>
  </a>
</div>
`;
